### PR TITLE
[KOGITO-6728] Processing multiple on-events

### PR DESCRIPTION
**Pull request by fjtirado**
Small one to take into account multiple onEvents (we were currently only considering the first one)
Notice the usage of exclusive: false in integration test

### CODE REVIEW

**Comment by kie-ci1** [comment]
**(tests) - kogito-runtimes job** `#1015` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 2480
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/1015/testReport/org.kie.kogito.integrationtests.quarkus/DecisionTableIT/___/">org.kie.kogito.integrationtests.quarkus.DecisionTableIT.(?)</a></summary>
java.lang.RuntimeException: <br/>java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors<br/>	[error]: Build step org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor#generateSources threw an exception: org.kie.kogito.codegen.rules.RuleCodegenError: Errors were generated during the code-generation process:<br/>java.lang.IllegalStateException: No data of this type found. Use containsData to check for this first.<br/><br/>	at org.kie.kogito.codegen.rules.IncrementalRuleCodegen.internalGenerate(IncrementalRuleCodegen.java:163)<br/>	at org.kie.kogito.codegen.core.AbstractGenerator.generate(AbstractGenerator.java:69)<br/>	at org.kie.kogito.codegen.core.ApplicationGenerator.lambda$generateComponents$0(ApplicationGenerator.java:97)<br/>	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)<br/>	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)<br/>	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)<br/>	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)<br/>	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)<br/>	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)<br/>	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)<br/>	at org.kie.kogito.codegen.core.ApplicationGenerator.generateComponents(ApplicationGenerator.java:99)<br/>	at org.kie.kogito.codegen.core.ApplicationGenerator.generate(ApplicationGenerator.java:74)<br/>	at org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor.generateFiles(KogitoAssetsProcessor.java:196)<br/>	at org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor.generateSources(KogitoAssetsProcessor.java:109)<br/>	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/>	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)<br/>	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/>	at java.base/java.lang.reflect.Method.invoke(Method.java:566)<br/>	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:882)<br/>	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)<br/>	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/>	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)<br/>	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)<br/>	at java.base/java.lang.Thread.run(Thread.java:829)<br/>	at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/>Caused by: java.lang.IllegalStateException: java.lang.IllegalStateException: No data of this type found. Use containsData to check for this first.<br/>	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)<br/>	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)<br/>	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)<br/>	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)<br/>	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:600)<br/>	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:678)<br/>	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:737)<br/>	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:919)<br/>	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)<br/>	at java.base/java.util.stream.IntPipeline.reduce(IntPipeline.java:496)<br/>	at java.base/java.util.stream.IntPipeline.max(IntPipeline.java:459)<br/>	at org.drools.modelcompiler.builder.PackageModel.generateRulesInUnit(PackageModel.java:703)<br/>	at org.drools.modelcompiler.builder.PackageModel.lambda$getRulesSource$6(PackageModel.java:591)<br/>	at java.base/java.util.concurrent.ConcurrentHashMap$KeySetView.forEach(ConcurrentHashMap.java:4698)<br/>	at org.drools.modelcompiler.builder.PackageModel.getRulesSource(PackageModel.java:582)<br/>	at org.drools.modelcompiler.builder.PackageModelWriter.<init>(PackageModelWriter.java:37)<br/>	at org.kie.kogito.codegen.rules.KogitoPackageSources.dumpSources(KogitoPackageSources.java:68)<br/>	at org.drools.modelcompiler.builder.ModelBuilderImpl.buildRules(ModelBuilderImpl.java:224)<br/>	at org.drools.modelcompiler.builder.ModelBuilderImpl.doSecondBuildStep(ModelBuilderImpl.java:118)<br/>	at org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:115)<br/>	at org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:99)<br/>	at org.kie.kogito.codegen.rules.IncrementalRuleCodegen.internalGenerate(IncrementalRuleCodegen.java:157)<br/>	... 24 more<br/>Caused by: java.lang.IllegalStateException: No data of this type found. Use containsData to check for this first.<br/>	at com.github.javaparser.ast.Node.getData(Node.java:505)<br/>	at com.github.javaparser.ast.CompilationUnit.getPrinter(CompilationUnit.java:154)<br/>	at com.github.javaparser.ast.Node.lambda$getPrinter$1(Node.java:199)<br/>	at java.base/java.util.Optional.map(Optional.java:265)<br/>	at com.github.javaparser.ast.Node.getPrinter(Node.java:199)<br/>	at com.github.javaparser.ast.Node.toString(Node.java:322)<br/>	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)<br/>	at java.base/java.util.TreeMap$ValueSpliterator.forEachRemaining(TreeMap.java:2890)<br/>	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)<br/>	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)<br/>	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:952)<br/>	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:926)<br/>	at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:327)<br/>	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)<br/>	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)<br/>	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)<br/>	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)<br/>	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)<br/>	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)<br/><br/>Caused by: java.lang.RuntimeException: <br/>io.quarkus.builder.BuildException: Build failure: Build failed due to errors<br/>	[error]: Build step org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor#generateSources threw an exception: org.kie.kogito.codegen.rules.RuleCodegenError: Errors were generated during the code-generation process:<br/>java.lang.IllegalStateException: No data of this type found. Use containsData to check for this first.<br/><br/>	at org.kie.kogito.codegen.rules.IncrementalRuleCodegen.internalGenerate(IncrementalRuleCodegen.java:163)<br/>	at org.kie.kogito.codegen.core.AbstractGenerator.generate(AbstractGenerator.java:69)<br/>	at org.kie.kogito.codegen.core.ApplicationGenerator.lambda$generateComponents$0(ApplicationGenerator.java:97)<br/>	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)<br/>	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)<br/>	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)<br/>	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)<br/>	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)<br/>	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)<br/>	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)<br/>	at org.kie.kogito.codegen.core.ApplicationGenerator.generateComponents(ApplicationGenerator.java:99)<br/>	at org.kie.kogito.codegen.core.ApplicationGenerator.generate(ApplicationGenerator.java:74)<br/>	at org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor.generateFiles(KogitoAssetsProcessor.java:196)<br/>	at org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor.generateSources(KogitoAssetsProcessor.java:109)<br/>	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/>	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)<br/>	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/>	at java.base/java.lang.reflect.Method.invoke(Method.java:566)<br/>	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:882)<br/>	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)<br/>	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/>	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)<br/>	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)<br/>	at java.base/java.lang.Thread.run(Thread.java:829)<br/>	at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/>Caused by: java.lang.IllegalStateException: java.lang.IllegalStateException: No data of this type found. Use containsData to check for this first.<br/>	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)<br/>	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)<br/>	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)<br/>	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)<br/>	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:600)<br/>	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:678)<br/>	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:737)<br/>	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:919)<br/>	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)<br/>	at java.base/java.util.stream.IntPipeline.reduce(IntPipeline.java:496)<br/>	at java.base/java.util.stream.IntPipeline.max(IntPipeline.java:459)<br/>	at org.drools.modelcompiler.builder.PackageModel.generateRulesInUnit(PackageModel.java:703)<br/>	at org.drools.modelcompiler.builder.PackageModel.lambda$getRulesSource$6(PackageModel.java:591)<br/>	at java.base/java.util.concurrent.ConcurrentHashMap$KeySetView.forEach(ConcurrentHashMap.java:4698)<br/>	at org.drools.modelcompiler.builder.PackageModel.getRulesSource(PackageModel.java:582)<br/>	at org.drools.modelcompiler.builder.PackageModelWriter.<init>(PackageModelWriter.java:37)<br/>	at org.kie.kogito.codegen.rules.KogitoPackageSources.dumpSources(KogitoPackageSources.java:68)<br/>	at org.drools.modelcompiler.builder.ModelBuilderImpl.buildRules(ModelBuilderImpl.java:224)<br/>	at org.drools.modelcompiler.builder.ModelBuilderImpl.doSecondBuildStep(ModelBuilderImpl.java:118)<br/>	at org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:115)<br/>	at org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:99)<br/>	at org.kie.kogito.codegen.rules.IncrementalRuleCodegen.internalGenerate(IncrementalRuleCodegen.java:157)<br/>	... 24 more<br/>Caused by: java.lang.IllegalStateException: No data of this type found. Use containsData to check for this first.<br/>	at com.github.javaparser.ast.Node.getData(Node.java:505)<br/>	at com.github.javaparser.ast.CompilationUnit.getPrinter(CompilationUnit.java:154)<br/>	at com.github.javaparser.ast.Node.lambda$getPrinter$1(Node.java:199)<br/>	at java.base/java.util.Optional.map(Optional.java:265)<br/>	at com.github.javaparser.ast.Node.getPrinter(Node.java:199)<br/>	at com.github.javaparser.ast.Node.toString(Node.java:322)<br/>	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)<br/>	at java.base/java.util.TreeMap$ValueSpliterator.forEachRemaining(TreeMap.java:2890)<br/>	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)<br/>	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)<br/>	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:952)<br/>	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:926)<br/>	at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:327)<br/>	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)<br/>	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)<br/>	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)<br/>	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)<br/>	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)<br/>	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)<br/><br/>Caused by: io.quarkus.builder.BuildException: <br/>Build failure: Build failed due to errors<br/>	[error]: Build step org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor#generateSources threw an exception: org.kie.kogito.codegen.rules.RuleCodegenError: Errors were generated during the code-generation process:<br/>java.lang.IllegalStateException: No data of this type found. Use containsData to check for this first.<br/><br/>	at org.kie.kogito.codegen.rules.IncrementalRuleCodegen.internalGenerate(IncrementalRuleCodegen.java:163)<br/>	at org.kie.kogito.codegen.core.AbstractGenerator.generate(AbstractGenerator.java:69)<br/>	at org.kie.kogito.codegen.core.ApplicationGenerator.lambda$generateComponents$0(ApplicationGenerator.java:97)<br/>	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:271)<br/>	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1655)<br/>	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)<br/>	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)<br/>	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)<br/>	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)<br/>	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)<br/>	at org.kie.kogito.codegen.core.ApplicationGenerator.generateComponents(ApplicationGenerator.java:99)<br/>	at org.kie.kogito.codegen.core.ApplicationGenerator.generate(ApplicationGenerator.java:74)<br/>	at org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor.generateFiles(KogitoAssetsProcessor.java:196)<br/>	at org.kie.kogito.quarkus.common.deployment.KogitoAssetsProcessor.generateSources(KogitoAssetsProcessor.java:109)<br/>	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/>	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)<br/>	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/>	at java.base/java.lang.reflect.Method.invoke(Method.java:566)<br/>	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:882)<br/>	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)<br/>	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/>	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)<br/>	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)<br/>	at java.base/java.lang.Thread.run(Thread.java:829)<br/>	at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/>Caused by: java.lang.IllegalStateException: java.lang.IllegalStateException: No data of this type found. Use containsData to check for this first.<br/>	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)<br/>	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)<br/>	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)<br/>	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)<br/>	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:600)<br/>	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:678)<br/>	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:737)<br/>	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:919)<br/>	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)<br/>	at java.base/java.util.stream.IntPipeline.reduce(IntPipeline.java:496)<br/>	at java.base/java.util.stream.IntPipeline.max(IntPipeline.java:459)<br/>	at org.drools.modelcompiler.builder.PackageModel.generateRulesInUnit(PackageModel.java:703)<br/>	at org.drools.modelcompiler.builder.PackageModel.lambda$getRulesSource$6(PackageModel.java:591)<br/>	at java.base/java.util.concurrent.ConcurrentHashMap$KeySetView.forEach(ConcurrentHashMap.java:4698)<br/>	at org.drools.modelcompiler.builder.PackageModel.getRulesSource(PackageModel.java:582)<br/>	at org.drools.modelcompiler.builder.PackageModelWriter.<init>(PackageModelWriter.java:37)<br/>	at org.kie.kogito.codegen.rules.KogitoPackageSources.dumpSources(KogitoPackageSources.java:68)<br/>	at org.drools.modelcompiler.builder.ModelBuilderImpl.buildRules(ModelBuilderImpl.java:224)<br/>	at org.drools.modelcompiler.builder.ModelBuilderImpl.doSecondBuildStep(ModelBuilderImpl.java:118)<br/>	at org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:115)<br/>	at org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:99)<br/>	at org.kie.kogito.codegen.rules.IncrementalRuleCodegen.internalGenerate(IncrementalRuleCodegen.java:157)<br/>	... 24 more<br/>Caused by: java.lang.IllegalStateException: No data of this type found. Use containsData to check for this first.<br/>	at com.github.javaparser.ast.Node.getData(Node.java:505)<br/>	at com.github.javaparser.ast.CompilationUnit.getPrinter(CompilationUnit.java:154)<br/>	at com.github.javaparser.ast.Node.lambda$getPrinter$1(Node.java:199)<br/>	at java.base/java.util.Optional.map(Optional.java:265)<br/>	at com.github.javaparser.ast.Node.getPrinter(Node.java:199)<br/>	at com.github.javaparser.ast.Node.toString(Node.java:322)<br/>	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)<br/>	at java.base/java.util.TreeMap$ValueSpliterator.forEachRemaining(TreeMap.java:2890)<br/>	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)<br/>	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)<br/>	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:952)<br/>	at java.base/java.util.stream.ReduceOps$ReduceTask.doLeaf(ReduceOps.java:926)<br/>	at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:327)<br/>	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)<br/>	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)<br/>	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)<br/>	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)<br/>	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)<br/>	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)<br/><br/>Caused by: org.kie.kogito.codegen.rules.RuleCodegenError: <br/>Errors were generated during the code-generation process:<br/>java.lang.IllegalStateException: No data of this type found. Use containsData to check for this first.<br/><br/>Caused by: java.lang.IllegalStateException: java.lang.IllegalStateException: No data of this type found. Use containsData to check for this first.<br/>Caused by: java.lang.IllegalStateException: No data of this type found. Use containsData to check for this first.<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.kogito-runtimes/1015/display/redirect

**Review by hbelmiro — APPROVED** [review]

**Comment by kie-ci1** [comment]
**(tests) - kogito-examples job** `#1110` was: **UNSTABLE**
Possible explanation: This should be test failures


**Test results:**
- PASSED: 432
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/1110/testReport/org.kie.kogito.examples/CallbackRestIT/Build_projects___testCallbackRest/">Build projects / org.kie.kogito.examples.CallbackRestIT.testCallbackRest</a></summary>
org.awaitility.core.ConditionTimeoutException: <br/>Assertion condition defined as a lambda expression in org.kie.kogito.examples.CallbackRestIT that uses java.lang.String 1 expectation failed.<br/>Expected status code <404> but was <200>.<br/> within 30 seconds.<br/>	at org.kie.kogito.examples.CallbackRestIT.testCallbackRest(CallbackRestIT.java:50)<br/>Caused by: java.lang.AssertionError: <br/>1 expectation failed.<br/>Expected status code <404> but was <200>.<br/><br/>	at org.kie.kogito.examples.CallbackRestIT.lambda$testCallbackRest$0(CallbackRestIT.java:55)<br/>
</details>

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-runtimes.tests.downstream.kogito-examples/1110/display/redirect

**Comment by fjtirado** [comment]
jenkins rerun kogito-examples tests

File: kogito-serverless-workflow/kogito-serverless-workflow-builder/src/main/java/org/kie/kogito/serverless/workflow/parser/handlers/EventHandler.java [hunk 1]
```diff
@@ -44,12 +44,20 @@ public void handleStart() {
...
-            nodes.add(filterAndMergeNode(factory, onEvent.getEventDataFilter(), isStartState ? ServerlessWorkflowParser.DEFAULT_WORKFLOW_VAR : getVarName(),
-                    (f, inputVar, outputVar) -> buildEventNode(f, onEventRef, inputVar, outputVar)));
-        }
+        return joinNodes(factory, state.getOnEvents().stream().map(onEvent -> processOnEvent(factory, onEvent)).collect(Collectors.toList()), state.isExclusive() ? Join.TYPE_XOR : Join.TYPE_AND);
+    }
+
+    private MakeNodeResult processOnEvent(RuleFlowNodeContainerFactory<?, ?> factory, OnEvents onEvent) {
+        MakeNodeResult result = joinNodes(factory,
+                onEvent.getEventRefs().stream().map(onEventRef -> filterAndMergeNode(factory, onEvent.getEventDataFilter(), isStartState ? ServerlessWorkflowParser.DEFAULT_WORKFLOW_VAR : getVarName(),
+                        (f, inputVar, outputVar) -> buildEventNode(f, onEventRef, inputVar, outputVar))).collect(Collectors.toList()),
```
Comment (ricardozanini) [inline kogito-serverless-workflow/kogito-serverless-workflow-builder/src/main/java/org/kie/kogito/serverless/workflow/parser/handlers/EventHandler.java:53]: nit; next time can you please add one step at a line? it's hard to read this way :)

**Review by ricardozanini — APPROVED** [review]

File: kogito-serverless-workflow/kogito-serverless-workflow-builder/src/main/java/org/kie/kogito/serverless/workflow/parser/handlers/EventHandler.java (hunk 1 above)
Comment (fjtirado) [reply kogito-serverless-workflow/kogito-serverless-workflow-builder/src/main/java/org/kie/kogito/serverless/workflow/parser/handlers/EventHandler.java:53]: should we not change the formatter to automatically do that? Im basically not using end of line on purpose to allow the formatter to do its job

Comment (ricardozanini) [reply kogito-serverless-workflow/kogito-serverless-workflow-builder/src/main/java/org/kie/kogito/serverless/workflow/parser/handlers/EventHandler.java:53]: Not sure if we have such a rule. If we don't, +1 to add it. :)

**Pull request merged**